src/uu/cp/src/cp.rs

```diff
@@ -898,9 +925,35 @@ fn copy_directory(root: &Path, target: &Target, options: &Options) -> CopyResult
     let mut hard_links: Vec<(String, u64)> = vec![];
 
     for path in WalkDir::new(root) {
-        let path = or_continue!(or_continue!(path).path().canonicalize());
+        let p = or_continue!(path);
+        let is_symlink = fs::symlink_metadata(p.path())?.file_type().is_symlink();
+        let path = if options.no_dereference && is_symlink {
+            // we are dealing with a symlink. Don't follow it
+            match env::current_dir() {
+                Ok(cwd) => cwd.join(resolve_relative_path(p.path())),
```

Comment:
I might be wrong, but doesn’t this prefix the current directory twice now (once in `resolve_relative_path()`, once here)?

Reply:
I had a look. 
`cwd.join(resolve_relative_path(p.path()))`and `cwd.join(normalize_path(p.path()))` return the same